extensions/cornerstone-dynamic-volume/src/panels/PanelGenerateImage.tsx

```diff
@@ -0,0 +1,414 @@
+            ...prevState[key],
+            ...generateOptions[key],
+          };
+        } else {
+          newState[key] = generateOptions[key];
+        }
+      });
+      return newState;
+    });
+  };
+
+  // Establish a reference to the viewer API context
+  const { activeViewportIndex, viewports } = viewportGridService.getState();
+  const displaySetInstanceUID = viewports[0].displaySetInstanceUIDs[0];
+  const volumeLoaderScheme = 'cornerstoneStreamingDynamicImageVolume'; // Loader id which defines which volume loader to use
+  const computedVolumeId = `cornerstoneStreamingImageVolume:${uuidComputedVolume}`;
+  const dynamicVolumeId = `${volumeLoaderScheme}:${displaySetInstanceUID}`;
+
+  useEffect(() => {
+    // ~~ Subscription
+    const added = cornerstoneViewportService.EVENTS.VIEWPORT_DATA_CHANGED;
+    const subscriptions = [];
+
+    // Used to make sure initialization of computed volume only runs once
+    let initStatus = false;
+
+    [added].forEach(evt => {
+      subscriptions.push(
+        cornerstoneViewportService.subscribe(evt, evtdetails => {
+          evtdetails.viewportData.data.forEach(volumeData => {
+            if (volumeData.volumeId.split(':')[0] === volumeLoaderScheme) {
+              if (!initStatus) {
+                initStatus = true;
+                setMyDynamicVolume(volumeData.volume);
+                const options = numTimePointsToOptions(
+                  volumeData.volume._numTimePoints
+                );
+                const range = [1, volumeData.volume._numTimePoints];
+                setTimeOptions(prevArray => [...prevArray, ...options]);
+                setRangeValues(prevArray => [...prevArray, ...range]);
+                setSliderValues(range);
+                setTimeFramesToUse([0, 1]);
+                const computedVolumeInit = createComputedVolume(
+                  volumeData.volume.volumeId,
+                  computedVolumeId
+                );
+                setMyComputedVolume(computedVolumeInit);
+              }
+            }
+          });
+        }).unsubscribe
+      );
+    });
+
+    return () => {
+      subscriptions.forEach(unsub => {
+        unsub();
+      });
+    };
+  }, []);
+
+  // Get computed volume from cache, calculate the data across the time frames,
+  // set the scalar data to the computedVolume, and create displaySet
+  function onGenerateImage() {
+    const computedVolume = cache.getVolume(computedVolumeId);
+    const dataInTime = cstUtils.dynamicVolume.generateImageFromTimeData(
+      myDynamicVolume,
+      generateOptions.Operation,
+      timeFramesToUse
+    );
+
+    // Add loadStatus.loaded to computed volume and set to true
+    computedVolume.loadStatus = {};
+    computedVolume.loadStatus.loaded = true;
+
+    // Set computed scalar data to volume
+    const scalarData = computedVolume.getScalarData();
+    for (let i = 0; i < dataInTime.length; i++) {
+      scalarData[i] = dataInTime[i];
+    }
+
```

Comment:
can't you just update the scalarData with dataInTime? instead of looping over?

Reply:
I don't believe so, I think I need to get the scalar data using `.getScalarData` and then assign it this was as setting it directly using `.scalarData = dataInTime` does not work